#167190401 Paginate article

Lundii opened the pull request:
#### What does this PR do?
Add pagination to getting articles by category
#### Description of Task to be completed?
- added pagination
#### How should this be manually tested?
- visit the staging app link https://ah-commando-react-pr-35.herokuapp.com/
#### Any background context you want to provide?

[e2e testing](https://drive.google.com/file/d/1FVlePMKtQNTAe5NiKNkDDQ-hu5SSmH5v/view?usp=sharing)

#### What are the relevant pivotal tracker stories?
[#167190401](https://www.pivotaltracker.com/story/show/167190401)

#### Screenshots (if appropriate)
<img width="1438" alt="Screenshot 2019-09-12 at 11 29 55 AM" src="https://user-images.githubusercontent.com/37886814/64777227-aa6f5100-d551-11e9-9b87-b31f2716dde4.png">

#### Questions:
N/A

Lundii committed:
ft(frontend): paginate category articles

- add pagination

[Finishes #167190401]

Hadeneekeh commented on .eslintrc:
```diff
@@ -46,7 +46,7 @@
     "no-param-reassign": 0,
     "comma-dangle": 2,
-    "no-console": 1,
+    "no-console": 0,
```
Great job so far. However, I think the team agreed to 1, not 0 for console log.

Lundii replied:
True. It was an oversight. I will update it now.

chuxmykel commented on src/views/App.js:
```diff
@@ -5,6 +5,7 @@ import Search from '@Views/SearchPage/SearchPage';
 import Home from '@Views/HomePage';
 import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';
 import SignUp from '@Components/Forms/SignUp';
+import CategoryPage from '@Views/CategoryPage';
```
Hey @Lundii. I will suggest that unless it's unavoidable, always place new lines of code below existing ones to avoid merge conflicts as much as possible.

Lundii replied:
Noted. it's just that EsLint flags that import that it should above some other imports in the file. That's why I placed it above.

Lundii committed:
ch(search): refactored search functionality

- added color to the pages
- rearranged the folder structure

[Finishes #168466772]

Hadeneekeh reviewed (approved):
Other than files like CategoryBody.scss and MainCardSection.scss that needs a new line, everything LGTM! Great job @Lundii

Lundii commented:
> Other than files like CategoryBody.scss and MainCardSection.scss that needs a new line, everything LGTM! Great job @Lundii

Noted. I will fix it. Thanks for the observation

Lundii commented:
> > Other than files like CategoryBody.scss and MainCardSection.scss that needs a new line, everything LGTM! Great job @Lundii
> 
> Noted. I will fix it. Thanks for the observation

Done

Lundii committed:
Minor Fix

chuxmykel reviewed (approved):
Looks good to me.

cvjude reviewed (approved):
Monday Can you make the footer go down, please.

Lundii committed:
minor fix

Lundii committed:
Merge branch 'ch/168466772-refactor-search-function' of https://github.com/andela/ah-commando-frontend into ft/167190401-paginate-article

Lundii committed:
Merge branch 'develop' into ft/167190401-paginate-article

cvjude reviewed (approved):
LGTM

Lundii commented:
> Monday Can you make the footer go down, please.

Alright. I have done that.

encodedBicoding reviewed (approved):
LGTM

chiboycalix reviewed (approved):
LGTM
